modules/charon-core/src/main/java/org/wso2/charon3/core/schema/ServerSideValidator.java

```diff
@@ -248,11 +252,55 @@ public static AbstractSCIMObject validateUpdatedSCIMObject(AbstractSCIMObject ol
             // Check for required attributes.
             validateSCIMObjectForRequiredAttributes(newObject, resourceSchema);
         }
+
+        Map<String, Attribute> attributes = validatedObject.getAttributeList();
+
+        for (Map.Entry<String, Attribute> entry : attributes.entrySet()) {
+            String key = entry.getKey();
+            Attribute value = entry.getValue();
+
+            if (value instanceof SimpleAttribute && StringUtils.equals(key, SCIMConstants.UserSchemaConstants.LOCALE)) {
+                String localeAttributeValue = ((SimpleAttribute) value).getValue().toString();
+
+                if (!isValidLocale(localeAttributeValue)) {
+                    throw new BadRequestException
+                            ("Provided locale value " + localeAttributeValue + " is invalid");
+                }
+            }
+        }
+
         // Check for schema list.
         validateSchemaList(validatedObject, resourceSchema);
         return validatedObject;
     }
 
+    private static boolean isValidLocale(String localeStr) {
+
+        if (localeStr == null || StringUtils.isEmpty(localeStr)) {
+            return false;
+        }
+
+        // Split the locale string into parts (language and country)
+        String[] parts = localeStr.split("-");
+
+        if (parts.length != 2) {
+            return false; // Must have exactly two parts: language and country
+        }
+
+        String language = parts[0];
+        String country = parts[1];
+
+        // Check if the locale is available in the system
+        for (Locale availableLocale : Locale.getAvailableLocales()) {
```

Comment:
Will looping through all the available locales impact the performance?

Reply:
As local update is done very rarely it will not impact. Also how many available locales which is getting iterated in this case.

Reply:
Seems there are more than 500 locales to iterate.

Reply:
As this is not a frequent update operation, and looping 500 elements will not take considerable time.